great_expectations/expectations/expectation.py

```diff
@@ -2373,6 +2374,13 @@ class BatchExpectation(Expectation, ABC):
     domain_type: ClassVar[MetricDomainTypes] = MetricDomainTypes.TABLE
     args_keys: ClassVar[Tuple[str, ...]] = ()
 
+    @pydantic.validator("mostly")
+    @classmethod
+    def validate_mostly(cls, mostly: float) -> float:
+        if not (0 <= mostly <= 1):
+            raise ValueError("mostly must be a value between 0 and 1")
+        return mostly
```

Comment:
Weirdly enough, `pydantic.Field` and the in-built validation done through `gt` and `lt` didn't mesh well with inheritance